pythonAPIClient/camera.py

```diff
@@ -14,8 +32,14 @@ class IPCamera(Camera):
     """
     # TODO: define extra retrieval attributes and constructor of ip camera object
     # replace others with desired field names
-    def __init__(self, others):
-        super(IPCamera, self).__init__(others)
+    def __init__(self, ip,port,brand,model,image_path,video_path):
```

Comment:
You need to pass all of the parent class  parameters in the derived class constructors.  This link should have some more helpful information! https://www.python-course.eu/python3_inheritance.php

super(IPCamera, self) gets the parent class, so for example in this line, you need to pass the parent class constructor parameters to the init function on line 36.